packages/eslint-plugin/lib/rules/no-unnecessary-type-assertion.js

```diff
@@ -0,0 +1,179 @@
+      return false; // if there are any other numeric properties, this is not a tuple
+    }
+  }
+  return true;
+}
+
+/**
+ *
+ * @param {Node} node node being linted
+ * @param {Context} context linting context
+ * @param {ts.TypeChecker} checker TypeScript typechecker
+ * @returns {void}
+ */
+function checkNonNullAssertion(node, context, checker) {
+  /**
+   * @type {ts.NonNullExpression}
+   */
+  const originalNode = context.parserServices.esTreeNodeToTSNodeMap.get(node);
+
+  if (!originalNode) {
+    return;
+  }
+  const type = checker.getTypeAtLocation(originalNode.expression);
+
+  if (type === checker.getNonNullableType(type)) {
+    context.report({
+      node,
+      message: FAILURE_STRING,
+      fix(fixer) {
+        return fixer.removeRange([
+          originalNode.expression.end,
+          originalNode.end
+        ]);
+      }
+    });
+  }
+}
+
+/**
+ * @param {Node} node node being linted
+ * @param {Context} context linting context
+ * @param {ts.TypeChecker} checker TypeScript typechecker
+ * @returns {void}
+ */
+function verifyCast(node, context, checker) {
+  /**
+   * @type {ts.AssertionExpression}
+   */
+  const originalNode = context.parserServices.esTreeNodeToTSNodeMap.get(node);
+
+  if (!originalNode) {
+    return;
+  }
+
+  const typesToIgnore = context.options[0];
+
+  if (
+    typesToIgnore &&
+    typesToIgnore.indexOf(originalNode.type.getText()) !== -1
+  ) {
+    return;
+  }
+  const castType = checker.getTypeAtLocation(originalNode);
+
+  if (
+    tsutils.isTypeFlagSet(castType, ts.TypeFlags.Literal) ||
+    (tsutils.isObjectType(castType) &&
+      (tsutils.isObjectFlagSet(castType, ts.ObjectFlags.Tuple) ||
+        couldBeTupleType(castType)))
+  ) {
+    // It's not always safe to remove a cast to a literal type or tuple
+    // type, as those types are sometimes widened without the cast.
+    return;
+  }
+
+  const uncastType = checker.getTypeAtLocation(originalNode.expression);
+
+  if (uncastType === castType) {
+    context.report({
+      node,
+      message: FAILURE_STRING,
+      fix(fixer) {
+        return originalNode.kind === ts.SyntaxKind.TypeAssertionExpression
+          ? fixer.removeRange([
+              originalNode.getStart(),
+              originalNode.expression.getStart()
+            ])
+          : fixer.removeRange([originalNode.expression.end, originalNode.end]);
+      }
+    });
+  }
+}
+
+/** @type {import("eslint").Rule.RuleModule} */
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Warns if a type assertion does not change the type of an expression',
+      category: 'TypeScript-specific',
+      recommended: false,
+      extraDescription: [util.tslintRule('no-unnecessary-type-assertion')],
+      url: util.metaDocsUrl('no-unnecessary-type-assertion')
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'array',
+        items: {
+          type: 'string'
+        }
+      }
+    ],
```

Comment:
usually its a good practice to use object for complex options, it will allow us to extend options more easily, and its easier to understand what option actually does

unless its simple option like string or number (eg. `always | never`)

Reply:
Changed to use an object.